File: backend/app/models/experience.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class ProjectItem(BaseModel):
    title: str
    description: str

# Request Models
class ExperienceCreate(BaseModel):
    company: str
    location: str
    position: str
    start_date: str
    end_date: str  # Can be "Present" or date
    projects: List[ProjectItem] = []

class ExperienceUpdate(BaseModel):
    company: Optional[str] = None
    location: Optional[str] = None
    position: Optional[str] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    projects: Optional[List[ProjectItem]] = None

# Response Models
class ExperienceResponse(BaseModel):
    id: str
    user_id: str
    company: str
    location: str
    position: str
    start_date: str
    end_date: str
    projects: List[ProjectItem] = []
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

File: frontend/src/api/experience.ts
import api from './axios'

export interface ProjectItem {
  title: string
  description: string
}

export interface ExperienceResponse {
  id: number
  user_id: number
  company: string
  location: string
  position: string
  start_date: string
  end_date: string
  projects: ProjectItem[]
  created_at: string
  updated_at: string
}

export interface ExperienceCreate {
  company: string
  location: string
  position: string
  start_date: string
  end_date: string
  projects?: ProjectItem[]
}

export interface ExperienceUpdate {
  company?: string
  location?: string
  position?: string
  start_date?: string
  end_date?: string
  projects?: ProjectItem[]
}

export const experienceApi = {
  getExperience: async (): Promise<ExperienceResponse[]> => {
    const response = await api.get<ExperienceResponse[]>('/api/experience/')
    return response.data
  },

  getExperienceById: async (experienceId: number): Promise<ExperienceResponse> => {
    const response = await api.get<ExperienceResponse>(`/api/experience/${experienceId}`)
    return response.data
  },

  createExperience: async (data: ExperienceCreate): Promise<ExperienceResponse> => {
    const response = await api.post<ExperienceResponse>('/api/experience/', data)
    return response.data
  },

  updateExperience: async (experienceId: number, data: ExperienceUpdate): Promise<ExperienceResponse> => {
    const response = await api.put<ExperienceResponse>(`/api/experience/${experienceId}`, data)
    return response.data
  },

  deleteExperience: async (experienceId: number): Promise<void> => {
    await api.delete(`/api/experience/${experienceId}`)
  },
}
